kombu/utils/json.py

```diff
@@ -80,6 +82,8 @@ def dumps(s, _dumps=json.dumps, cls=None, default_kwargs=None, **kwargs):
 
 def object_hook(dct):
     """Hook function to perform custom deserialization."""
+    if "__datetime__" in dct:
+        return parse_iso8601(dct["datetime"])
```

Comment:
I fully agree we should migrate to the stdlib version and remove this backport completely from the codebase.

Reply:
The idea behind that was to have the same behavior, as in celery, but you are totally right that it makes sense to be based on internal functions